tools/lsp/language.rs

```diff
@@ -1115,25 +1287,102 @@ fn get_code_lenses(
+    let doc = document_cache.get_document(&text_document.uri)?;
+    let version = document_cache.document_version(&text_document.uri);
 
-        let inner_components = doc.inner_components.clone();
+    let mut result = vec![];
 
-        let mut r = vec![];
+    if cfg!(any(feature = "preview-builtin", feature = "preview-external")) {
+        let inner_components = doc.inner_components.clone();
 
         // Handle preview lens
-        r.extend(inner_components.iter().filter(|c| !c.is_global()).filter_map(|c| {
+        result.extend(inner_components.iter().filter(|c| !c.is_global()).filter_map(|c| {
             Some(CodeLens {
                 range: util::node_to_lsp_range(&c.root_element.borrow().debug.first()?.node),
                 command: Some(create_show_preview_command(true, &text_document.uri, c.id.as_str())),
                 data: None,
             })
         }));
+    }
 
-        Some(r)
-    } else {
+    if let Some(node) = &doc.node {
+        let has_non_ws_token = node
+            .children_with_tokens()
+            .any(|nt| nt.kind() != SyntaxKind::Whitespace && nt.kind() != SyntaxKind::Eof);
+        if !has_non_ws_token {
+            if let Some(range) = populate_command_range(&node) {
+                result.push(CodeLens {
+                    range: range.clone(),
+                    command: Some(create_populate_command(
+                        text_document.uri.clone(),
+                        version.clone(),
+                        "Start with Hello World!".to_string(),
+                        r#"
+import { AboutSlint, Button, VerticalBox } from "std-widgets.slint";
+
+export component Demo {
+    VerticalBox {
+        alignment: start;
+        Text {
+            text: "Hello World!";
+            font-size: 24px;
+            horizontal-alignment: center;
+        }
+        AboutSlint {
+            preferred-height: 150px;
+        }
+        HorizontalLayout { alignment: center; Button { text: "OK!"; } }
+    }
+}
+"#
+                        .to_string(),
+                    )),
+                    data: None,
+                });
+                result.push(CodeLens {
+                    range,
+                    command: Some(create_populate_command(
+                        text_document.uri.clone(),
+                        version.clone(),
+                        "Start with Window".to_string(),
+                        r#"
+export component MainWindow inherits Window {
+    title: "Slint Main Window";
+
+    min-width: 200px;
+    min-height: 200px;
+}
+"#
+                        .to_string(),
+                    )),
+                    data: None,
+                });
+                result.push(CodeLens {
+                    range,
+                    command: Some(create_populate_command(
+                        text_document.uri.clone(),
+                        version.clone(),
+                        "Start with fixed size Window".to_string(),
```

Comment:
not sure we need that.  I don't think we should confuse the user with too many choices

Reply:
I think I added that round the time I initially did this PR since we had two people asking about how to do a fixed size window. I never saw that question again since then, so let's drop this.

Reply:
I only left a short version of "Hello World" in place.